hivemq-edge/src/main/java/com/hivemq/protocols/ProtocolAdapterMetrics.java

```diff
@@ -0,0 +1,34 @@
+package com.hivemq.protocols;
+
+import com.codahale.metrics.MetricRegistry;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.hivemq.metrics.HiveMQMetrics.ADAPTERS_CURRENT;
+import static com.hivemq.metrics.HiveMQMetrics.PROTOCOL_ADAPTER_PREFIX;
+
+@Singleton
+public class ProtocolAdapterMetrics {
+
+    private final @NotNull MetricRegistry metricRegistry;
+    private final AtomicInteger currentAdapters = new AtomicInteger();
```

Comment:
Why the explicit atomic?
Codahale has metricRegistry..counter() which provides the very same functionality.

Reply:
we wanted it to be a Gauge to be consistent with the other metrics. This is why we added the AtomicInteger although a counter would do the exact same job.